src/components/MoneyRequestHeader.tsx

```diff
@@ -209,7 +221,7 @@ function MoneyRequestHeader({session, parentReport, report, parentReportAction,
                 cancelText={translate('common.cancel')}
                 danger
             />
-            {isSmallScreenWidth && shouldShowHoldMenu && (
+            {shouldUseNarrowLayout && shouldShowHoldMenu && (
```

Comment:
The ProcessMoneyRequestHoldMenu is to be shown for small screen devices. However, since `shouldUseNarrowLayout` is used here, this will also get shown when the held report is opened in RHP thereby resulting in issue #42007